quilt/tools/check_functions.py

```diff
@@ -0,0 +1,112 @@
+import re
+import pandas as pd
+from pandas import DataFrame as df
+import numpy
+
+# defined as lowecase globals so importing the library results in clean syntax
+# like this:  qc.data[colname], qc.env, qc.filename, etc.
+filename = None                 # pylint:disable=C0103
+data = None                     # pylint:disable=C0103
+env = 'default'                 # pylint:disable=C0103
+seed = 0  # for reproducibility             # pylint:disable=C0103
+
+class CheckFunctionsReturn(Exception):
+    def __init__(self, result):
+        super(CheckFunctionsReturn, self).__init__()
+        self.result = result
+
+class CheckFunctionsException(Exception):
+    def __init__(self, result):
+        super(CheckFunctionsException, self).__init__()
+        self.result = result
+
+def check(expr, envs=None):
+    global env                  # pylint:disable=C0103
+    if envs is not None:
+        # TODO: error if env not found
+        expr = envs.get(env, expr)
+    if not expr:
+        raise CheckFunctionsReturn(expr)
+
+def print_recnums(msg, expr, maxrecs=30):
+    matching_recs = [str(i) for i, val in enumerate(expr) if val]
+    print('{}: {}{}'.format(msg, ",".join(matching_recs[0:maxrecs]),
+                            "" if len(matching_recs) < maxrecs else ",..."))
+
+def data_sample(*args, **kwargs):
+    global data, seed           # pylint:disable=C0103
+    if 'seed' in args:
+        seed = kwargs['seed']
+    if 'random_state' not in args:
+        kwargs['random_state'] = numpy.random.random_sample # pylint:disable=E1101
+    data = data.sample(*args, **kwargs)
+
+def check_column_enum(colrx, lambda_or_listexpr, envs=None):
+    if envs not in [None, 'default']:
+        check_column_regexp(colrx, envs[env])
+    for colname in list(data):
+        if re.search(colrx, colname):
+            if callable(lambda_or_listexpr):
+                check(lambda_or_listexpr(data[colname]))
+            else:
+                check(data[colname].isin(lambda_or_listexpr).all())
+
+VALRANGE_FUNCS = {
+    'mean':     lambda col, minval, maxval: check(minval <= col.mean() <= maxval),
```

Comment:
I'd move `check` out of the function, and leave just the `lambda ...: minval <= ... <= maxval`.

(You could be really fancy make it a map of names to methods, e.g. `'mean': ColumnType.mean` - but I guess that doesn't work for `abs`? Anyways, just a thought - not sure that makes it better.)

Reply:
DONE - also removed the minval/maxval